src/solver/optimisation/constraints/RampingIncreaseRate.cpp

```diff
@@ -0,0 +1,47 @@
+#include "RampingIncreaseRate.h"
+
+void RampingIncreaseRate::add(int pays, int cluster, int clusterIndex, int pdt, bool Simulation)
+{
+    if (!Simulation)
+    {
+        const PALIERS_THERMIQUES& PaliersThermiquesDuPays = problemeHebdo->PaliersThermiquesDuPays[pays];
+        double maxUpwardPowerRampingRate = PaliersThermiquesDuPays.maxUpwardPowerRampingRate[clusterIndex];
+        double pminDUnGroupeDuPalierThermique = PaliersThermiquesDuPays.pminDUnGroupeDuPalierThermique[clusterIndex];
+        // constraint : P(t) - P(t-1) - R^+ * M(t) - l * M^+(t) < 0
+        // logs.info() << "add up ramping constraint for cluster " << cluster << " pdt " << pdt;
```

Comment:
to be removed

Reply:
Keep the 1st line of comments, remove the 2nd

Reply:
thank you, never delete a formula ;)